packages/components/src/custom-select-control-v2/custom-select.tsx

```diff
@@ -88,11 +88,13 @@ function _CustomSelect(
 		label,
 		size,
 		store,
+		className,
 		...restProps
 	} = props;
 
 	return (
-		<>
+		// Where should `restProps` be forwarded to?
```

Comment:
This is something unrelated to this PR, but that we should definitely discuss. Currently, `...restProps` are assumed to be intended for `CustomSelectButton`, but I'd argue that folks would expect that most attributes would be assigned to the root wrapper (which will likely become `BaseControl`)

Reply:
I think it should be the `CustomSelectButton`. Is there anything other than `className` that you'd want to pass to the root wrapper? I'm thinking about refs and event listeners, which would pass more naturally onto the button.

Reply:
I'm worried that if we start adding exceptions like `className`, devEx will suffer. Also, if a consumer of the component wanted to set a `className` to the button, how would they do it?

An alternative approach could be to diverge V2 and V2 legacy further, so that they each provide a root wrapper 

Another though: if we don't pass _any_ props to the root wrapper element, do you see that as a limitation for consumers? For example,:
- let's say that as a consumer I want to hide `CustomSelectControl` — setting `display: none` would only hide the button, but keep the label around
- let's say I'm rendering `CustomSelectControlV2` inside a flexbox parent, and I'd like to set `flex-shring` on it. Using `className` or `style` won't cut it, since those styles should be applied to the root wrapper. The only way to achieve that will be to add an external wrapper component to `CustomSelectControlV2`.

I think that this is a broader conversation that we need to have as we export components that internally compose smaller components.

For example, in `DropdownMenuV2` we solved this by exporting a separate `Trigger` subcomponent, and exposing a `trigger` render prop. Which means that props by default can be passed to the root element, while trigger-specific props are passed directly to the `Trigger` subcomponent.

Curious to hear @DaniGuardiola 's thoughts on this one too.